Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp

```diff
@@ -300,13 +310,26 @@ WebCore::DisplayRefreshMonitor& ThreadedCompositor::displayRefreshMonitor() cons
 void ThreadedCompositor::frameComplete()
 {
     ASSERT(m_compositingRunLoop->isCurrent());
+    displayUpdateFired();
     sceneUpdateFinished();
 }
 
 void ThreadedCompositor::targetRefreshRateDidChange(unsigned rate)
 {
     ASSERT(RunLoop::isMain());
-    m_displayRefreshMonitor->setTargetRefreshRate(rate);
+
+    if (!rate)
+        rate = c_defaultRefreshRate;
+    m_display.displayUpdate = { 0, rate / 1000 };
```

Comment:
I see displayUpdate is always used in the compositing thread except here. Should we use `m_compositingRunLoop->performTask` to do this in the compositing thread instead?

Reply:
Makes sense, `performTaskSync`.